garglk/wingrid.cpp

```diff
@@ -77,8 +77,13 @@ void win_textgrid_rearrange(window_t *win, rect_t *box)
 
     for (k = 0; k < dwin->height; k++) {
         touch(dwin, k);
-        std::fill(&dwin->lines[k].chars[dwin->width], dwin->lines[k].chars.end(), ' ');
-        for (auto *attr = dwin->lines[k].attrs.begin() + dwin->width; attr != dwin->lines[k].attrs.end(); ++attr) {
+        // std::fill(&dwin->lines[k].chars[dwin->width], dwin->lines[k].chars.end(), ' ');
```

Comment:
What's the rationale for changing `std::fill` here?  Is this a bug in MSVC that fails with the fill, or is it a bug in Gargoyle that just so happens to work with libstdc++?

Reply:
The Microsoft STL does not like `std::fill` with one pointer and one `std::array<>::iterator`. [cppreference suggests](https://en.cppreference.com/w/cpp/algorithm/fill) that the first and second arguments to `fill` should be the same type, so I'm leaning towards libstdc++ being more lenient than the MS STL.

For reference, this is the compile error:

```
C:\Users\Adrian\Documents\Development\Cpp\garglk\garglk\wingrid.cpp(80,9): error: no matching function for call to 'fill'
   80 |         std::fill(&dwin->lines[k].chars[dwin->width], dwin->lines[k].chars.end(), ' ');
      |         ^~~~~~~~~
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.37.32822\include\xutility(5032,19): note: candidate template ignored: deduced conflicting types for parameter '_FwdIt' ('unsigned int *' vs. 'iterator' (aka '_Array_iterator<unsigned int, 256ULL>'))
 5032 | _CONSTEXPR20 void fill(const _FwdIt _First, const _FwdIt _Last, const _Ty& _Val) {
      |                   ^
```

Reply:
I got curious and looked it up, and libstdc++ really just `typedef`s a pointer to be the iterator, hence the original code worked there. The MS STL has a distinct iterator type, so it breaks. Same goes for comparing pointers to iterators.

Anyways, your adjusted `std::fill` call works.